kernel/src/client/default/mod.rs

```diff
@@ -1,6 +1,6 @@
 //! # Default Engineinterface
 //!
-//! The default implementation of [`Engineinterface`] is [`DefaultEngineInterface`].
+//! The default implementation of [`EngineInterface`] is [`DefaultEngineInterface`].
```

Comment:
should we harmonize the name? EngineClient or EngineInterface? 
I feel we should harmonize the java side as well - TableClient to EngineClient?

Reply:
We spent quiet a bit of time discussing this and settled on `EngineInterface`. At least in the rust world, it really is just an "interface", in that it, by default, doesn't supply any functionality. 

So I'd vote for changing the name on the java side ;)

Reply:
sounds good!